### ISSUE TITLE
[FFmpeg] No audio playback on 4-channel audio device

### ISSUE THREAD

**Issue by gorskii** [telegramdesktop/tdesktop#8183]
I have a usb audio output device which has a 4-channel 44,1k 24bit soundcard in it. It is a DJ controller (Pioneer DDJ-400) which uses standard Windows Audio drivers and has no options to configure it's output configuration.
It works well as a system default output device, any application can use this device to play audio.
But if I try to play some audio in Telegram Desktop media player, the player does nothing: I hear no audio, playhead is not moving.
If I play a video file, the video is playing, but the audio track is not.
If I set default output to internal speakers, audio is playing after I close player bar and hit the play button again.

In the log.txt file there are some error messages written when I press play button:
```
[2020.06.30 20:36:13] Streaming Error: Error in avcodec_find_decoder (code: 8, text: Error number 8 occurred).
[2020.06.30 20:36:13] Audio Info: recreating audio device and reattaching the tracks
[2020.06.30 20:36:13] Audio Context Error: 40961, Invalid Device
```

Not sure if it's an OpenAl, ffmpeg or TDesktop issue

### Steps to reproduce
1. Set 4-channel USB audio device as a system default output device
2. Select any audio file in any chat and press play

### Expected behaviour
Audio should play through channels 1 and 2 of a 4 channel USB audio device.

### Actual behaviour
No audio is playing. `Streaming Error: Error in avcodec_find_decoder (code: 8, text: Error number 8 occurred).` and `Audio Context Error: 40961, Invalid Device` messages appear in the app's log.

### Configuration
**Operating system:**
win10 Pro x64 build 10.0.18362

**Version of Telegram Desktop:**
2.1.13

**Logs:**
[log.txt](https://github.com/telegramdesktop/tdesktop/files/4853634/log.txt)

**Comment by gorskii** [comment]
Just tried to play some files through VLC media player which also uses FFmpeg, and it crashed with my USB sound card selected as an output device. So it is definitely an FFmpeg related issue.
I will post a bug in FFmpeg bugtracker and close this issue as soon as it will be solved.

**Comment by gorskii** [comment]
Also tested on Ubuntu 16.04 and macOS 10.14.6 - no problems with audio output on this device neither in TDesktop, nor in VLC. So I consider this issue to be Windows related only.

**Comment by ilya-fedin** [comment]
> Just tried to play some files through VLC media player which also uses FFmpeg

tdesktop uses ffmpeg just for decoding, audio output works only with the help of OpenAL

**Comment by gorskii** [comment]
> tdesktop uses ffmpeg only for decoding, audio output works only with the help of OpenAL

Could it be a problem during decoding process? I suppose that `Streaming Error: Error in avcodec_find_decoder (code: 8, text: Error number 8 occurred).` means that there is an error in the process of finding suitable decoder. I also think that the cause of this error could be some incorrect device properties provided by OpenAL.

Is there any chance to enable extensive logging of OpenAL or FFmpeg modules to narrow down the scope of this issue? I have tried to enter `debugmode` to generate debug logs, but haven't found any audio-related messages different from those I have posted in the description.
